ranges.go

```diff
@@ -0,0 +1,52 @@
+package lipgloss
+
+import (
+	"strings"
+
+	"github.com/charmbracelet/x/ansi"
+)
+
+// StyleRanges allows to, given a string, style ranges of it differently.
+// The function will take into account existing styles.
+func StyleRanges(s string, ranges []Range) string {
+	if len(ranges) == 0 {
+		return s
+	}
+
+	var buf strings.Builder
+	lastIdx := 0
+	stripped := ansi.Strip(s)
+
+	// Use Truncate and TruncateLeft to style match.MatchedIndexes without
+	// losing the original option style:
+	for _, rng := range ranges {
+		// Add the text before this match
+		if rng.Start > lastIdx {
+			buf.WriteString(ansi.Cut(s, lastIdx, rng.Start))
+		}
+		if l := len(stripped); rng.End >= l {
+			rng.End = l - 1
+		}
+		// Add the matched range with its highlight
+		buf.WriteString(rng.Style.Render(stripped[rng.Start : rng.End+1]))
```

Comment:
My concern here is dealing with multibyte runes and multi-rune characters. Not sure what's the behavior with such cases.

Reply:
right, yeah, probably is not correct indeed